### PR TITLE
`kornia.nerf` improvements

### PR DESCRIPTION
**Pull request by edgarriba**
#### Changes

- add a class `NerfModelRenderer` to render single views from a pinhole model camer and add tests
- make some api improvements to the `NerfSolver` class
- add `kornia.nerf` to the docs
- rename `rays.py` -> `samplers.py`
- rename `renderer.py` -> `volume_renderer.py`

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [x] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/nerf/nerf_solver.py [hunk 1]
```diff
@@ -139,34 +194,44 @@ def _train_one_epoch(self) -> float:
         Returns:
             Average psnr over all epoch rays
         """
-        if self._cameras is None:
-            raise TypeError("The camera should be a PinholeCamera. Gotcha None. You init the training before train?")
-
+        KORNIA_CHECK(self._nerf_model is not None, "The model should be a NeRF model. Gotcha None.")
```
Comment (cjpurackal) [inline kornia/nerf/nerf_solver.py:197]: gotcha? too casual? XD

**Review by cjpurackal — APPROVED** [review]

File: kornia/nerf/nerf_solver.py [hunk 2]
```diff
@@ -96,10 +105,56 @@ def init_training(
...
+            lr,
+        )
 
     @property
-    def nerf_model(self) -> Module:
+    def nerf_model(self) -> Module | None:
+        """Returns the NeRF model."""
         return self._nerf_model
 
     # FIXME: Remove this function - consistency is checked in
```
Comment (YanivHollander) [inline kornia/nerf/nerf_solver.py:160]: This commented function can be removed

File: kornia/nerf/nerf_solver.py [hunk 3]
```diff
@@ -182,11 +247,15 @@ def run(self, num_epochs: int = 1) -> None:
                 current_time = datetime.now().strftime("%H:%M:%S")  # noqa: DTZ005
                 print(f"Epoch: {i_epoch}: epoch_psnr = {epoch_psnr}; time: {current_time}")
 
-    def render_views(self, cameras: PinholeCamera) -> ImageTensors:
+        # delete data
+        del self._imgs
+
+    def render_views(self, cameras: PinholeCamera, batch_size: int = 4096) -> ImageTensors:
```
Comment (YanivHollander) [inline kornia/nerf/nerf_solver.py:253]: @edgarriba, is this function needed given your addition of `NerfModelRenderer`?

**Review by YanivHollander — COMMENTED** [review]
Great additions. See my below comments

**Pull request merged**